### ISSUE TITLE
Flat version of hash tree

### ISSUE THREAD

**Issue by Un3qual** [mceachen/closure_tree#270]
Is there a way to get a flat version of a hash tree. It doesn't have to be a hash tree, but just be ordered in the same way, and be flat.

**Comment by mceachen** [comment]
besides `selfAndDescendants`?

**Comment by Un3qual** [comment]
Yeah. I could use that but my model is Comments and a Post has_many
Comments, so i would have to have 1 level of nesting

**Comment by Un3qual** [comment]
Also, self_and_descendants is ordered by depth, so the order is wrong.
